ios/RNMagicScript/components/UiNodeSelector.swift

```diff
@@ -32,31 +32,21 @@ class UiNodeSelector: NodeSelecting  {
 
     func hitTest(ray: Ray) -> HitTestResult? {
         // Check if any component has intercepted focus (e.g. DropdownListNode)
-        if let node = focusInterceptedNodes().first {
-            if let hitResult = node.hitTest(ray: ray) {
-                return hitResult
-            }
-            return (node: node, point: .zero)
+        let focusInterceptedNodes = collectFocusInterceptedNodes()
+        let focusInterceptedHitResults: [HitTestResult] = focusInterceptedNodes.compactMap { $0.hitTest(ray: ray) }
+        if let hitResult = getClosestHitResult(focusInterceptedHitResults, ray: ray) {
+            return hitResult
+        }
+        
+        // If there is no hitResults for focus-intercepted nodes, we still want to return any focus-intercepted node
+        if let firstNode = focusInterceptedNodes.first {
+            return (node: firstNode, point: .zero)
         }
         
         // Check hit test in a standard way
-        var hitResults: [HitTestResult] = []
         let topNodes: [BaseNode] = rootNode.childNodes.compactMap { $0 as? BaseNode }
```

Comment:
will this code ever but hit?  I think it will only be hit if `focusInterceptedNodes` is empty.

Reply:
That's the goal of this code:
- if there is any component which has intercepted the focus (ex. dropdown List is showing its list), then return that component (so that it can handle the input: either select list item or dismiss the list)
- else do standard **hitTest**